Mathlib/CategoryTheory/Join/Opposites.lean

```diff
@@ -0,0 +1,157 @@
+/-
+Copyright (c) 2025 Robin Carlier. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Robin Carlier
+-/
+import Mathlib.CategoryTheory.Join.Basic
+import Mathlib.CategoryTheory.Opposites
+
+/-!
+# Opposites of joins of categories
+
+This file constructs the canonical equivalence of categories `(C ⋆ D)ᵒᵖ ≌ Dᵒᵖ ⋆ Cᵒᵖ`.
+The equivalence gets characterized in both directions.
+
+-/
+
+namespace CategoryTheory.Join
+open Opposite
+
+variable (C D : Type*) [Category C] [Category D]
+
+/-- The forward direction of `Join.opEquiv`. -/
+def opEquivFunctor : (C ⋆ D)ᵒᵖ ⥤ Dᵒᵖ ⋆ Cᵒᵖ :=
+  Functor.leftOp <|
+    Join.mkFunctor (inclRight _ _).rightOp (inclLeft _ _).rightOp ({app _ := (edge _ _).op})
```

Comment:
```suggestion
    Join.mkFunctor (inclRight _ _).rightOp (inclLeft _ _).rightOp {app _ := (edge _ _).op}
```
untested, and I'm not sure if this will work